pkg/idp/third_party.go

```diff
@@ -55,14 +55,14 @@ type Configuration struct {
 	// AuthURL is the authorize url, typically something like: https://example.org/oauth2/auth
 	// Should only be used when the OAuth2 / OpenID Connect server is not supporting OpenID Connect Discovery and when
 	// `provider` is set to `generic`.
-	// validate that this field is required only when Provider field == "generic"
-	AuthURL string `json:"auth_url" yaml:"auth_url" validate:"required_if=Provider generic"`
+	// validate that this field is required only when Provider field == "generic" and IssuerURL is not empty
```

Comment:
Shouldn't this be `when Provider field == "generic" and IssuerURL is empty` (if issuerURL is not empty then we can (?) assume that discovery is supported

Reply:
I got confused, you're 100% right. Fixing it